### ISSUE TITLE
"minimal" mode

### ISSUE THREAD

**Issue by jywarren** [publiclab/leaflet-environmental-layers#123]
We should make it possible to show lots of layers in a minimal mode with small dots you can click on, kind of like these screenshots:

![image](https://user-images.githubusercontent.com/24359/53528335-59b57080-3ab7-11e9-9c9c-169f5b7348d0.png)

![image](https://user-images.githubusercontent.com/24359/53528496-bb75da80-3ab7-11e9-833a-996ae80b9018.png)

We could color them differently, and it could allow us to show more layers if we wanted to. They could be colored based on the listing shown in this mockup: https://github.com/publiclab/leaflet-environmental-layers/issues/60#issuecomment-460462757

Once this mode is off, we could re-show the larger icons for everything!

**Comment by avsingh999** [comment]
@jywarren @gauravano  can you explain more?
thank you : )

**Comment by jywarren** [comment]
Basically looking for a way to turn all the markers into smaller, standardized "dots" of different colors, rather than their full icons. So that we can show more layers without filling up the screen!

**Comment by avsingh999** [comment]
Okay, we will have to change all layer Icon into small dots. Am I right or not @jywarren .
Thank you : )

**Comment by jywarren** [comment]
Yes, but we would want this to be toggle-able, so we'd want to preserve the
previous icon, too!

**Comment by avsingh999** [comment]
@jywarren  I want to try to solve this issue.

**Comment by jywarren** [comment]
that'd be great!

**Comment by rkpattnaik780** [comment]
@jywarren what about using a floating action button to toggle between marker icons and dots?

**Comment by jywarren** [comment]
This could work, but actually we may want to make a general-purpose
JavaScript method to do the toggling, and then demonstrate it with a button
(but apart from the demo, leave it to whoever's using the library to decide
how this will be triggered). Make sense? Thank you for your input!!!

**Comment by crisner** [comment]
After looking at different ways of doing this, I think we can use leaflet's circle marker to simplify layer markers and render them on canvas instead of creating separate DOM elements for each marker to improve performance. I tried this out in EonetFiresLayer and this is how it looks:
![127 0 0 1_5500_example_index html (7)](https://user-images.githubusercontent.com/29401459/72345918-d9adee80-36fa-11ea-8181-02d527c53e73.png)

I am planning to match the styling to the ones shown in the first comment to give it a more 'standardized' look.  From what I see we will have edit each layer to conditionally choose the right marker depending upon the mode but would try to check if there is a better way of doing this.

@jywarren and @sagarpreet-chadha, what do you think?

**Comment by jywarren** [comment]
Oh this is really nice, thanks! One note on performance is that although
I'm sure Leaflet has plenty of ways to manage complexity and make canvas
markers a bit simpler to work with, but generally DOM markers are handled
pretty efficiently by the browser until a few hundred, and are worth it
because the browser handles events, whereas with canvas the event handling
is written in JavaScript. That got me on the unearthing layer, but since
that has 15k markers, there was no other way to do it-- i used the glify
library and so that was on a GPU-accelerated 3d canvas.

None of this changes your decision which I think is great -- just trying to
provide some extra context for this sort of decisionmaking, having worked
with lots of markers in the past! Thank you!

**Comment by crisner** [comment]
Thanks! Any cons I should be aware of when using canvas for rendering markers? What kind of issues did you have? I looked at glify but it looked like it takes in only geojson data.

**Comment by sagarpreet-chadha** [comment]
Hey,

So fetching markers icon from some external URL will take time. 
Using L.circle i think will remove this latency.

I have one quick question, why do you want to use Canvas?

**Comment by crisner** [comment]
> So fetching markers icon from some external URL will take time.
Using L.circle i think will remove this latency.

You are right. That's why I decided to go with `L.circleMarker`.

> I have one quick question, why do you want to use Canvas?

Well, since we do not have a lot of markers in layers other than open sense and air quality index this may be redundant. I was hoping to reduce the page lag because of the number of marker elements in the dom in case the user turns on multiple layers and especially if one of those layers turn out to be layers like air quality index. What do you think? Should we deal with performance issues as a separate issue?

**Comment by sagarpreet-chadha** [comment]
Hey,

So fetching markers icon from some external URL will take time. 
Using L.circle i think will remove this latency.

I have one quick question, why do you want to use Canvas?

**Comment by sagarpreet-chadha** [comment]
Okay yes makes sense.
Can you try using L.circle for heavy layers and see if this still lags? (I am hoping this should not lag!)

Otherwise if it lags, then it makes sense to invest time in Canvas. Thanks!

**Comment by crisner** [comment]
Awesome. By the way, just to be clear, according to the docs L.circle is an obsolete way to instantiate circles. I will be using L.circleMarker which extends path and instantiates a circle marker.

**Comment by jywarren** [comment]
I found that after 500-1000 markers, DOM markers slowed down a lot on slower devices. I had to actually modify glify (and had meant to submit the changes upstream, need to make a todo for this) because there's some complexity to how to determine which points to send a "click" to when someone clicks on a canvas. I think it's a clustering algorithm... thing. I fixed it in the library but it hurt my brain a bit. I think if you use a standard library it should be OK, and this should be handled for you!
